website/src/components/mdx/tabs/tab-groups.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+
+type Groups = {
+  [group: string]: string;
+};
+
+type SetGroup = (groupId: string, value: string) => void;
+
+type TabGroupContextShape = {
+  groups: Groups;
+  setGroup: SetGroup;
+};
+
+const TabGroupContext = createContext<TabGroupContextShape>({
+  groups: {},
+  setGroup: () => {},
+});
+
+const getLocalStorageKey = (groupId: string) => `tab-${groupId}`;
+
+export const TabGroupProvider: FC = ({ children }) => {
+  const [groups, setGroups] = useState<Groups>({});
+
+  const handleSetGroup: SetGroup = (groupId, value) => {
+    if (groups[groupId] === value) return;
+
+    setGroups({ ...groups, [groupId]: value });
+
+    if (typeof window !== "undefined" && window.localStorage) {
+      window.localStorage.setItem(getLocalStorageKey(groupId), value);
+    }
+  };
+
+  return (
+    <TabGroupContext.Provider value={{ groups, setGroup: handleSetGroup }}>
+      {children}
+    </TabGroupContext.Provider>
+  );
+};
+
+type TabGroupReturn = [string, (value: string) => void];
+
+export function useActiveTab(
+  defaultValue: string,
+  groupId?: string
+): TabGroupReturn {
+  const { groups, setGroup } = useContext(TabGroupContext);
+  const [activeTab, setActiveTab] = useState(defaultValue);
+
+  if (!groupId) return [activeTab, setActiveTab];
+
+  const activeGroupTab = useMemo(() => {
+    let value: string | null = groups[groupId];
+
+    if (!value && typeof window !== "undefined" && window.localStorage) {
+      value = window.localStorage.getItem(getLocalStorageKey(groupId));
```

Comment:
@rstaib I don't have much experience when it comes to SSR/SSG, how could I make it so the selection is actually read from the `localStorage` once it's running in the client's browser? At the moment it always starts out with the `defaultValue` after the first access, subsequent client-routed requests work as expected.